spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/EmbeddedCassandra.scala

```diff
@@ -43,60 +46,81 @@ trait EmbeddedCassandra {
-  val rpcPorts = getNodePropertySeq(RpcPortProperty).asInstanceOf[IndexedSeq[Int]]
-
-  private def getNodePropertySeq(nodeProperty: NodeProperty): IndexedSeq[Any] = {
-    nodeProperty match {
-      case NativePortProperty => getValueSeq(NativePortProperty.propertyName).map(e => e.toInt)
-      case RpcPortProperty => getValueSeq(RpcPortProperty.propertyName).map(e => e.toInt)
-      case HostProperty => getValueSeq(HostProperty.propertyName).map(e => InetAddress.getByName(e))
-      case _ => throw new RuntimeException("Wrong node input property")
-    }
+
+  trait TypedProperty {
+    type ValueType
+    def convertValueFromString(str: String): ValueType
+    def checkValueType(obj: Any): ValueType
+  }
+
+  trait IntProperty extends TypedProperty {
+    type ValueType = Int
+    def convertValueFromString(str: String) = str.toInt
+    def checkValueType(obj: Any) =
+      obj match {
+        case x: Int => x
+        case _ => throw new ClassCastException (s"Expected Int, got ${obj.getClass.getName}")
+      }
+  }
+
+  trait InetAddressProperty extends TypedProperty {
+    type ValueType = InetAddress
+    def convertValueFromString(str: String) = InetAddress.getByName(str)
+    def checkValueType(obj: Any) =
+      obj match {
+        case x: InetAddress => x
+        case _ => throw new ClassCastException (s"Expected InetAddress, got ${obj.getClass.getName}")
+      }
   }
 
-  private def getValueSeq(propertyName: String) = {
+  abstract sealed class NodeProperty(val propertyName: String) extends TypedProperty
+  case object HostProperty extends NodeProperty("IT_TEST_CASSANDRA_HOSTS") with InetAddressProperty
+  case object NativePortProperty extends NodeProperty("IT_TEST_CASSANDRA_NATIVE_PORTS") with IntProperty
+  case object RpcPortProperty extends NodeProperty("IT_TEST_CASSANDRA_RPC_PORTS") with IntProperty
+
+  private def getValueSeq(propertyName: String): Seq[String] = {
     sys.env.get(propertyName) match {
       case Some(p) => p.split(",").map(e => e.trim).toIndexedSeq
       case None => IndexedSeq()
     }
   }
 
-  def getProperty(nodeProperty: NodeProperty) = {
-    nodeProperty match {
-      case p: PropertyType => sys.env.get(p.propertyName)
-      case _ => throw new RuntimeException("Wrong node input property")
-    }
-  }
+  private def getValueSeq(nodeProperty: NodeProperty): Seq[nodeProperty.ValueType] =
+    getValueSeq(nodeProperty.propertyName).map(x => nodeProperty.convertValueFromString(x))
+
+  val hosts = getValueSeq(HostProperty)
+  val nativePorts = getValueSeq(NativePortProperty)
+  val rpcPorts = getValueSeq(RpcPortProperty)
+
+  def getProperty(nodeProperty: NodeProperty): Option[String] =
+    sys.env.get(nodeProperty.propertyName)
+  
+  def getPropertyOrThrowIfNotFound(nodeProperty: NodeProperty): String =
+    getProperty(nodeProperty).getOrElse(
+      throw new ConfigurationException(s"Missing ${nodeProperty.propertyName} in system environment"))
 }
```

Comment:
I am not sure about this, a pattern where an env or system setting missing will throw an exception.
Settings should be options with sensible defaults if not found so that deploys can move through a chain of cluster environments like dev/test/staging etc with no code changes.